lib/Runtime/Types/RecyclableObject.h

```diff
@@ -26,7 +26,7 @@ namespace Js {
         InlineCache* inlineCache;
         PolymorphicInlineCache * polymorphicInlineCache;
         FunctionBody * functionBody;
-        PropertyString * propertyString;
+        PropertyRecordUsageCache* propertyRecordUsageCache; // Interior pointer to a Symbol or PropertyString. Okay as long as PropertyValueInfo is stack-allocated.
```

Comment:
>Interior pointer [](start = 63, length = 16)

This seems moderately creepy; I'm open to other suggestions for how to represent a thing that's either a Symbol or a PropertyString.

Reply:
What keeps the Symbol/PropertyString alive? You might want either an additional pointer to the DynamicObject that owns the PropertyRecordUsageCache or make this a union of Symbol & PropertyString?

---
In reply to: [162213674](https://github.com/Microsoft/ChakraCore/pull/4569#discussion_r162213674) [](ancestors = 162213674)

Reply:
I like your idea of keeping an extra pointer to the owning object, thanks! It seems that PropertyValueInfo is always stack-allocated, and you mentioned that stack references are checked for interior pointers, but relying on that behavior feels very brittle.

---
In reply to: [162227539](https://github.com/Microsoft/ChakraCore/pull/4569#discussion_r162227539) [](ancestors = 162227539,162213674)